src/features/filter.ts

```diff
@@ -1,10 +1,48 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetFilterAction = { type: 'filter/STATUS', payload: Status };
+type SetQueryAction = { type: 'filter/QUERY', payload: string };
+
+type Action = SetFilterAction | SetQueryAction;
+
+const setFilter = (status: Status): SetFilterAction => ({
+  type: 'filter/STATUS', payload: status,
+});
+
+const setQuery = (query: string): SetQueryAction => ({
+  type: 'filter/QUERY',
+  payload: query,
+});
+
+export const actions = { setQuery, setFilter };
+
+type State = {
+  query: string,
+  status: Status,
+};
+
+const initialState = {
+  query: '',
+  status: 'all' as Status,
+};
+
+const filterReducer = (filter: State = initialState, action: Action) => {
+  switch (action.type) {
+    case 'filter/STATUS':
+      return {
+        ...filter,
+        status: action.payload,
+      };
+
+    case 'filter/QUERY':
+      return {
+        ...filter,
+        query: action.payload.toLowerCase(),
```

Comment:
```suggestion
        query: action.payload,
```
it could be unexpected for the user
you might just do it when filtering todos